repos/known-good.proj

```diff
@@ -32,6 +32,7 @@
         <!-- Tier 1 -->
         <RepositoryReference Include="application-insights" />
         <RepositoryReference Include="aspnet-xdt" />
+        <RepositoryReference Include="sourcelink" />
```

Comment:
Adding this to the all-up proj is fine to get it building, but to support building from other root project nodes I'd expect a reference from some other project, maybe the Arcade repo?

Reply:
Actually sourcelink depends on Arcade and if I add it to arcade proj it might result in cyclic dependency. And that may become a problem. I think roslyn depends on it. But not sure about that.

Reply:
My assumption there was that the intention of adding sourcelink is to include it as a tool in the Arcade SDK, but I am lacking context. The dependency of `sourcelink.proj` on `arcade.proj` could be removed and the prebuilt/N-1 Arcade SDK used, AFAIK.

Reply:
I tried to remove it in the beginning. But then the build errors out saying that it was expecting arcade directory which it couldn't find. Later after adding arcade as a repoRef, the error vanished.

Reply:
Maybe this was still in there, which tries to use a source-built Arcade SDK:

```
  <ItemGroup>
    <UseSourceBuiltSdkOverride Include="@(ArcadeSdkOverride)" />
  </ItemGroup>
```

Either way, no point in changing this without knowing the intended usage.

Reply:
If there are no repo references for this repo, I'm not sure why we're adding it.  It would be good to sync with @tmat, since he requested it, to understand what the intention is for this and get this worked out before merging this PR.

Reply:
SourceLink should be referenced by all repos via Arcade SDK. Currently Arcade SDK references a [specific version](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/DefaultVersions.props#L93) of the package.

The package is referenced here: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets#L7-L10 (currently excluded when building from source, a condition we want to remove).

I think we should flow SourceLink to Arcade like we do Roslyn compiler package, so that all repos get the latest SourceLink with Arcade update.

Reply:
I think Arcade uses a frozen version of Roslyn, do we want to do the same for SourceLink?  I do think it makes sense to flow it using a Version.Details.xml entry and subscription and everything instead, we'll have to make some Arcade changes for that.

Reply:
Not sure what do you mean by "frozen". Arcade SDK captures the version of Roslyn that flows into Arcade repo here: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/Microsoft.DotNet.Arcade.Sdk.csproj#L71, so that everyone using that Arcade SDK build gets that version.

we should do the same for SourceLink.

Reply:
That means Arcade repo would subscribe for SourecLink updates.

Reply:
There's no Roslyn in [Version.Details.xml](https://github.com/dotnet/arcade/blob/0c2c6f3121c2c1fa5a4391b72a4df13ddc9e9959/eng/Version.Details.xml) though, and the last few updates at least to that variable are manual PRs (like https://github.com/dotnet/arcade/pull/3146), how is it getting updates?

Reply:
> how is it getting updates?

Apparently manually at this point. Seems like we are missing subscription of Arcade repo to some Roslyn channel.

Reply:
Whether the update is manual or automatically flows is secondary. The important thing is that if it is updated it will flow to all repos where Arcade SDK flows.

Reply:
> The package is referenced here: https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Arcade.Sdk/tools/RepositoryInfo.targets#L7-L10 (currently excluded when building from source, ***a condition we want to remove***).

@tmat Does SourceLink require source-control?  I'm not clear about including SourceLink with source-build.  What will happen when we build offline and there is no source-control?

Reply:
It does not require the full `.git` dir, just a couple of files that can be created by a script. See
https://github.com/dotnet/sourcelink/tree/master/docs#minimal-git-repository-metadata

Reply:
Cool, I will add that as part of my daily build work.